source/bson-corpus/binary-vector-subtype.md

```diff
@@ -0,0 +1,94 @@
+bear the additional cost of storing and processing vector data.
+
+`*` The early addition of the "Packed Bit" representation was to facilitate partnerships in the expanding market of
+Generative AI, specifically in Vector Quantization. Succinctly put, a Binary Quantized Vector is just a vector of 0s and
+1s (bits), but it is often represented as a list of uint8 (int in Python). So, for example, the vector `[255, 0]` would
+be shorthand for the 16 bit vector `[1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0]`.\
+The authors are well-aware of the inherent
+ambiguity here. This is a market-standard, unfortunately. Change is inevitable.
+
+## META
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Specification
+
+This spec introduces a new BSON binary subtype, the vector, with value `"\x09"`. Each vector can take one of multiple
+data types (dtypes). The following table lists the first dtypes implemented.
+
+| Vector data type | Alias      | Bits per vector element | [PyArrow Data Type](https://arrow.apache.org/docs/cpp/api/datatype.html) (for illustration) |
+| ---------------- | ---------- | ----------------------- | ------------------------------------------------------------------------------------------- |
+| `0x03`           | INT8       | 8                       | INT8                                                                                        |
+| `0x27`           | FLOAT32    | 32                      | FLOAT                                                                                       |
+| `0x10`           | PACKED_BIT | 1     `*`               | BOOL                                                                                        |
+
+As not all data types have a bit length equal to a multiple of 8, and hence do not fit squarely into a certain number of
+bytes, a second piece of metadata, the "padding" is included. This instructs the driver of the number of bits in the
+final byte that are to be ignored.
+
+The binary structure the vector subtype's value is this. Following the binary subtype `0x09` is a two-element byte
+array.
+
+- The first byte (dtype) describes its data type, such as float32 or int8. The table above shows the implemented
+  initially implemented in Python. The complete list of data types runs from `0x02` to `0x4b`
+
+- The second byte (padding) prescribes the number of bits to ignore in the final byte of the value.Ω
+
+- The remainder contains the actual vector elements packed according to dtype.
+
+All values use the little-endian format.
+
+## Reference Implementation
+
+Please consult the Python driver's `pymongo.binary` module. Prose tests described below can be found in
+`test.test_bson.TestBSON.test_vector`.
```

Comment:
Please add a link to the PYTHON ticket.

Reply:
I didn't see an example of a ticket, but I did add a link to the github module.

Reply:
done.